File: src/main/java/com/github/kmizu/yapp/SemanticException.java
package com.github.kmizu.yapp;

public class SemanticException extends RuntimeException {
  private static final long serialVersionUID = -3752445565800942042L;
  private int line;
  private int column;
  
  public SemanticException(Position pos, String message) {
    super(message);
    this.line   = pos.line();
    this.column = pos.column();
  }
  
  public String getErrorMessage() {
    return line + ":" + column + ":" + getMessage();
  }
}

File: src/main/scala/com/github/kmizu/yapp/SemanticException.scala
package com.github.kmizu.yapp

@SerialVersionUID(-3752445565800942042L)
case class SemanticException(pos: Position, message: String) extends RuntimeException(message) {
  val line  : Int = 0
  val column: Int = 0

  def getErrorMessage: String = {
    line + ":" + column + ":" + getMessage
  }
}